react_basic/src/pages/NewPost.jsx

```diff
@@ -0,0 +1,139 @@
+import { useRef, useState, useContext, useEffect } from 'react';
+import { PageWrapperContext } from '../pages/PageWrapper.jsx';
+import Loader from '../components/Loader.jsx';
+import validationService from '../services/validationService.js';
+import postService from '../services/postService.js';
+import windowUtils from '../services/windowUtils.js';
+import '../styles/NewPost.css';
+
+function NewPost() {
+  const { setErrorMsg, setSuccessMsg } = useContext(PageWrapperContext);
+
+  const [isLoading, setIsLoading] = useState(false);
+  const [errors, setErrors] = useState({});
+
+  const newPostFormRef = useRef();
+
+  useEffect(() => {
+    windowUtils.toggleWindowScroll(!isLoading);
+  }, [isLoading]);
+
+  function setTextareaHeight(e) {
+    const textarea = e.currentTarget;
+
+    const borderWidth = parseFloat(
+      getComputedStyle(textarea).getPropertyValue('border-width'),
+    );
+
+    textarea.style.height = '';
+    textarea.style.height = textarea.scrollHeight + borderWidth * 2 + 'px';
+  }
+
+  async function handleFormSubmit(e) {
```

Comment:
From my side of view such complex and heavy functions are very difficult to understand and read. Also this function contain double stIsLoading(true) calls, but this is inefficient. It's better to divide it into logical functions:
![image](https://github.com/user-attachments/assets/7e6aecd1-3f01-4e55-8ad2-1763bea529d1)